### ISSUE TITLE
EDSM commander name/API key not found

### ISSUE THREAD

**Issue by dmon82** [EDCD/EDDI#1290]
## EDDI version in which issue is found
3.4.0

## VoiceAttack version in which issue is found (as applicable)
1.7.5

## Steps to reproduce
In the EDSM responder configuration tab, enter API key and commander name, click the Obtain EDSM log button, button responds with "EDSM error received: Commander  name/API Key not found".

## Expected
Successful retrieval of EDSM log.

## Observed
Says that command name or API key was not found.

## Investigation
I'm using the same API key and commander name in the EDDB Market Collector, which successfully submits data that I can see in my flight log on EDSM.

**Comment by Tkael** [comment]
Are there any special / non-standard characters in your commander name?

**Comment by dmon82** [comment]
> 
> 
> Are there any special / non-standard characters in your commander name?

Nope, commander name is [cynictrash](https://www.edsm.net/en/user/profile/id/46282/cmdr/cynictrash). Nothing specific in the log file either, also just says that either one doesn't exist. I tried refreshing the API key (creating a new one), and the new key didn't work for me either.

**Comment by Tkael** [comment]
Hmm. I wonder if something is blocking the connection from EDDI to EDSM. Have you tried the same credentials on a different PC / network connection?

**Comment by dmon82** [comment]
I don't have another PC or connection, but the ED Market Collector as I said, does use the same credentials successfully. Unfortunately I can't compile the source myself to run it in the IDE for debugging, getting a lot of errors.

**Comment by Tkael** [comment]
Errors like?

**Comment by dmon82** [comment]
Ah, I'm on Windows 7 and Visual Studio 2013, my crap is too old heh.

**Comment by Tkael** [comment]
Ah. Yes, I'm on Windows 10 and Visual Studios Community Edition 2017.
Is your EDSM name the same as your in-game name?

**Comment by dmon82** [comment]
I should really check if I can use newer .NET versions at all on Windows 7 and get a new VS Express, there's a lot of neat neat features in newer C# versions of the language. But yes, CMDR name is cynictrash in one word on both, EDSM and in-game, and the ED Market Collector keeps successfully using the credentials and submits to [my EDSM profile](https://www.edsm.net/en/user/profile/id/46282/cmdr/cynictrash).

Not really sure what else to tell you, I even copy & pasted the same credentials from the EDMC program into the EDDI text input fields, and used the credentials in a POST request on https://www.edsm.net/api-commander-v1/get-credits in a quick test program. I can attach the EDDI log, but it doesn't tell anything other than the commander name or API key wasn't found. But I swear to the sun and stars, I'm using those credentials in the text fields.

[eddi_issue.zip](https://github.com/EDCD/EDDI/files/3146604/eddi_issue.zip)

**Comment by dmon82** [comment]
I've started integrating EDDI into my VA profile, and I think I have new information. The error only occurs when I run EDDI as the stand-alone application. I've run the configuration context from within VA, and this way it successfully authenticates with EDSM.

**Comment by Tkael** [comment]
Interesting. After it successfully authenticates with VA running, are you able to use the same credentials in standalone?

**Comment by Tkael** [comment]
Ok. Well, since 
1. we're not sure how to replicate the problem, 
2. it seems to be resolved, and
3. nobody else has reported the same issue
we're going to ahead and close this ticket.
o7 cmdr!

**Comment by arancaytar** [comment]
I have the same problem, but before I install VoiceAttack, I'd like to know: *Could* this be caused by a special character in my commander name? It has a space.

**Comment by Tkael** [comment]
It's unlikely that a space would cause the issue. One easy thing to check is whether your in-game name matches your EDSM account commander name. If the EDSM commander name differs from the in-game name, you need to make sure the EDSM responder has the EDSM commander name.

**Comment by cburschka** [comment]
I've tried different combinations to no effect - omitting the name when it matched, explicitly entering it in the client, then changing it on EDSM (to replace the space with an underscore) and entering it in the client accordingly.

Is there a log file I should look at? I'll open a new issue, and I guess I can get the Voice Attack trial version to see if I can fix it like OP did.

**Comment by arancaytar** [comment]
I've tried different combinations to no effect - omitting the name when it matched, explicitly entering it in the client, then changing it on EDSM (to replace the space with an underscore) and entering it in the client accordingly.

Is there a log file I should look at? I'll open a new issue, and I guess I can get the Voice Attack trial version to see if I can fix it like OP did.

**Comment by Tkael** [comment]
The most recent log file would be at `%appdata%/EDDI` and would be called `eddi.log`. I'd be happy to take a look at it if you post it (preferably in a new issue but referencing this one).

**Comment by arancaytar** [comment]
Okay, we can add another "mysterious resolution". I didn't install voice attack or anything else, other than restarting my computer. I can upload some log files if it might help shed more light; otherwise I guess "have you tried turning it off and on again" is good advice to anyone else who finds this issue.

**Comment by CptHarlock** [comment]
Hi, i would deeply apreciate if you can help me. I believe i have similar issue. My cmdr name is :Harlock: can this be the reason i cant get the conection between EDDI and EDSM to work? I get the commander/api key not found error.

**Comment by Tkael** [comment]
Is that your in-game commander name or your EDSM name? They are not necessarily the same. Make sure that the name matches your EDSM settings page. https://www.edsm.net/en/settings/api 

As above, it would be best to create a new issue and share your logs so that we can study this in more details.

**Comment by willneedit** [comment]
Encountered the issue as well, with a freshly created EDSM account and a new API key. Restarting EDDI (was standalone, restarted as standalone) fixed the issue for me.

Looks like updating the credentials in the text edit fields didn't update some neccessary internal states.

**Comment by Tkael** [comment]
Re-opening to see if we can replicate based on new information above.

**Comment by cloudymusic** [comment]
I just experienced the same issue as willneedit. On a fresh install of EDDI (standalone without VA), I typed in my EDSM API key and Commander name several times (as well as leaving the Commander Name field blank), and the log import would always fail with the error "Commander name/API Key not found." The log files indicated "StarMapService:sendEvent EDSM responded with Commander name/API Key not found" every time this happened.

However, I can confirm that like willneedit, filling in the EDSM API key and EDSM Commander Name fields properly, then quitting EDDI, re-launching, and then clicking the log import button _did_ cause the import to work properly.
